presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -407,4 +407,10 @@ public PlanNode visitValues(ValuesNode node, RewriteContext<Void> context)
                     rewrittenRows);
         }
     }
+
+    public boolean isCorrelated(Query subquery)
+    {
+        PlanNode subqueryPlan = createRelationPlan(subquery).getRoot();
```

Comment:
`createRelationPlan` has side-effects, so this is probably not a very good idea to do in a method that appears to be side effect-free.